### PR TITLE
docs: suggested impression data reference doc restructure

### PR DESCRIPTION
**Pull request by thomasheartman**
Here's my suggestion for changes to the impression data reference documentation. This is mostly just a restructuring of what was already there, with an added table, but I've also updated the SDK compatibility matrix and the create toggle API endpoint documentation.

I've got a couple questions that need to be resolved, but I'll mark those inline.

### CODE REVIEW

File: website/sidebars.js [hunk 1]
```diff
@@ -246,7 +246,7 @@ module.exports = {
                         'user_guide/activation_strategy',
                         'advanced/archived_toggles',
                         'advanced/audit_log',
-                        'advanced/impression_data',
+                        'advanced/impression-data',
```
Comment (thomasheartman) [inline website/sidebars.js:249]: I've changed it from an underscore to a hyphen. As of today, there's no clear standard to follow in this code base (neither for URLs nor for file names), but I'd like to standardize everything on using hyphens, so I've changed the URL here. That'll need to be updated in the front end too.

File: website/docusaurus.config.js [hunk 2]
```diff
@@ -160,6 +160,10 @@ module.exports = {
                         to: '/user_guide/activation_strategy',
                         from: '/user_guide/control_rollout',
                     },
+                    {
+                        from: '/advanced/impression_data',
```
Comment (thomasheartman) [inline website/docusaurus.config.js:164]: If we update the front end before merging the docs pr, we can do without the redirect, but it's probably no big deal either way.

File: website/sidebars.js (hunk 1 above)
Comment (FredrikOseberg) [reply website/sidebars.js:249]: Can we keep it as underscore and revisit this later? We've started merging the refactorings into the frontend at this point.

File: website/docs/advanced/impression-data.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+- [`getVariant` events](#example-getvariant)
+
+The `getVariant` event contains all the information found in an `isEnabled` event in addition to extra data that's only relevant to `getVariant` calls.
+
+This table describes all the properties on the impression events:
+
+| Property name | Description                                                                          | Event type               |
+|---------------|--------------------------------------------------------------------------------------|--------------------------|
+| `eventType`   | The type of the event: `isEnabled` or `getVariant`                                   | All                      |
+| `eventId`     | A globally unique id (GUID) assigned to this event.                                  | All                      |
```
[the remark this answers is not in the document]
Comment (FredrikOseberg) [reply website/docs/advanced/impression-data.md:29]: No, it's not mapped to anything inside of unleash. Purely for the users benefit.

File: website/docusaurus.config.js (hunk 2 above)
Comment (FredrikOseberg) [reply website/docusaurus.config.js:164]: Let's keep the redirect here until we finish the redux refactoring.

**Review by FredrikOseberg — APPROVED** [review]
LGTM

**Comment by thomasheartman** [comment]
Sweet! I'll merge it into your branch then, and we can take it from there. It resolves all my comments, at least, so we should be able to merge yours right in too, I think 😄

**Pull request merged**